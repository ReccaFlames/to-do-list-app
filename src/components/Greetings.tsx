import styled from "styled-components";

interface GreetingsProps {
    name: string;
    tasksCounter: number; 
}

const Greetings = (props: GreetingsProps) => {
    const { name, tasksCounter } = props;
    return (
        <div>
            <Title>Hello {name}! ðŸ‘‹</Title>
            <Subtitle>{tasksCounter} tasks for you Today</Subtitle>
        </div>
    );
}

export default Greetings;

const Title = styled.h1`
    margin: 1rem 0 .5rem 0;
`;

const Subtitle = styled.h4`
    margin: 0 0 2rem 0;
    color: #656565;
    font-weight: 400;
`;