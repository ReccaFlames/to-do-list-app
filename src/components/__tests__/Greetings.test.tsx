import { render } from '@testing-library/react';
import Greetings from '../Greetings';

describe('Greetings component', () => {

    it('render correctly', () => {
        const props = {
            name: 'Joe',
            tasksCounter: 2,
        } 
        
        const {baseElement} = render(<Greetings {...props}/>);

        // expect(getByText("Hello Joe! ðŸ‘‹")).toBeVisible();
        expect(baseElement).toMatchSnapshot();
    });
});