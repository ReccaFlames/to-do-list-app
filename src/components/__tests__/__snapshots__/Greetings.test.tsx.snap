// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greetings component render correctly 1`] = `
<body>
  <div>
    <div>
      <h1
        class="sc-bczRLJ gGXxBv"
      >
        Hello 
        Joe
        ! ðŸ‘‹
      </h1>
      <h4
        class="sc-gsnTZi cIBuFQ"
      >
        2
         tasks for you Today
      </h4>
    </div>
  </div>
</body>
`;
